extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/JpaJandexScavenger.java

```diff
@@ -121,70 +122,86 @@ private static void enlistExplicitClasses(IndexView index, JpaEntitiesBuildItem
                 domainObjectCollector.addEntity(className);
             }
         }
+
+        throwIfUnIndexedClassesNotEmpty(unIndexedClasses);
     }
 
     private static void enlistReturnType(IndexView index, JpaEntitiesBuildItem domainObjectCollector,
             Set<String> enumTypeCollector) {
         Collection<AnnotationInstance> annotations = index.getAnnotations(EMBEDDED);
-        if (annotations != null && annotations.size() > 0) {
-            for (AnnotationInstance annotation : annotations) {
-                AnnotationTarget target = annotation.target();
-                DotName jpaClassName = null;
-                switch (target.kind()) {
-                    case FIELD:
-                        // TODO could fail if that's an array or a generic type
-                        jpaClassName = target.asField().type().name();
-                        break;
-                    case METHOD:
-                        // TODO could fail if that's an array or a generic type
-                        jpaClassName = target.asMethod().returnType().name();
-                        break;
-                    default:
-                        throw new IllegalStateException("[internal error] @Embedded placed on a unknown element: " + target);
-                }
-                addClassHierarchyToReflectiveList(index, domainObjectCollector, enumTypeCollector, jpaClassName);
+        if (annotations == null) {
+            return;
+        }
+
+        Set<DotName> unIndexedClasses = new HashSet<>();
+        for (AnnotationInstance annotation : annotations) {
+            AnnotationTarget target = annotation.target();
+            DotName jpaClassName;
+            switch (target.kind()) {
+                case FIELD:
+                    // TODO could fail if that's an array or a generic type
+                    jpaClassName = target.asField().type().name();
+                    break;
+                case METHOD:
+                    // TODO could fail if that's an array or a generic type
+                    jpaClassName = target.asMethod().returnType().name();
+                    break;
+                default:
+                    throw new IllegalStateException("[internal error] @Embedded placed on a unknown element: " + target);
             }
+            addClassHierarchyToReflectiveList(index, domainObjectCollector, enumTypeCollector, jpaClassName,
+                    unIndexedClasses);
         }
+
+        throwIfUnIndexedClassesNotEmpty(unIndexedClasses);
```

Comment:
I don't think the pattern is right. You should collect all of them and throw only one exception. Here you collect 2 different lists. For me, this should be done only once in `discoverModelAndRegisterForReflection`.

Reply:
Good catch.